fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java

```diff
@@ -277,15 +281,39 @@ public List<String> listPartitionNamesByValue(String catalogName, String dbName,
         return ImmutableList.copyOf(partitionNames.build());
     }
 
+    public Statistics getTableStatisticsFromInternalStatistics(Table table, Map<ColumnRefOperator, Column> columns) {
+        List<ColumnRefOperator> requiredColumnRefs = new ArrayList<>(columns.keySet());
+        List<String> columnNames = requiredColumnRefs.stream().map(col -> columns.get(col).getName()).collect(
+                Collectors.toList());
+        List<ConnectorTableColumnStats> columnStatisticList =
+                GlobalStateMgr.getCurrentStatisticStorage().getConnectorTableStatistics(table, columnNames);
+
+        Statistics.Builder statistics = Statistics.builder();
+        for (int i = 0; i < requiredColumnRefs.size(); ++i) {
+            ColumnRefOperator columnRef = requiredColumnRefs.get(i);
+            ConnectorTableColumnStats connectorTableColumnStats = columnStatisticList.get(i);
+            if (connectorTableColumnStats != null) {
+                statistics.addColumnStatistic(columnRef, connectorTableColumnStats.getColumnStatistic());
+                statistics.setOutputRowCount(connectorTableColumnStats.getRowCount());
+            }
+        }
+        return statistics.build();
+    }
+
     public Statistics getTableStatistics(OptimizerContext session,
                                          String catalogName,
                                          Table table,
                                          Map<ColumnRefOperator, Column> columns,
                                          List<PartitionKey> partitionKeys,
                                          ScalarOperator predicate) {
-        Optional<ConnectorMetadata> connectorMetadata = getOptionalMetadata(catalogName);
-        return connectorMetadata.map(metadata ->
-                metadata.getTableStatistics(session, table, columns, partitionKeys, predicate)).orElse(null);
+        Statistics statistics = getTableStatisticsFromInternalStatistics(table, columns);
```

Comment:
If there is unknown, could we use Internal estimated columns + connector column ？

Reply:
I am worried that the inconsistent collection methods will lead to worse plans